pkg/auth/providers/ldap/ldap_provider.go

```diff
@@ -141,6 +145,47 @@ func (p *ldapProvider) GetPrincipal(principalID string, token v3.Token) (v3.Prin
 		return v3.Principal{}, err
 	}
 
+	if p.groupSearchProvider() {
+		config, caPool, err := p.getLDAPConfig()
+		if err != nil {
+			logrus.Warnf("ldap search principals failed to get ldap config: %s\n", err)
+			return v3.Principal{}, nil
+		}
+
+		lConn, err := ldap.Connect(config, caPool)
+		if err != nil {
+			logrus.Warnf("ldap search principals failed to connect to ldap: %s\n", err)
+			return v3.Principal{}, nil
+		}
+		defer lConn.Close()
+
+		if scope == p.userScope {
+			userPrincipals, err := p.searchUser(externalID, config, lConn)
+			if err != nil {
+				return v3.Principal{}, err
+			}
+
+			if len(userPrincipals) != 1 {
+				return v3.Principal{}, fmt.Errorf("get principal for %s returned %d results",
+					externalID, len(userPrincipals))
+			}
+			return userPrincipals[0], nil
```

Comment:
we should call `isThisUserMe` on the user

Reply:
yup done